In [1]:
# probability estimations
p_positive = (.98 * .005) + (.1 *.995)
print("Probability of positive result: " + str(p_positive))
correct_diagnosis = .98
print("Probability of correct diagnosis (sufferer): " + str(correct_diagnosis))
correct_diagnosis_non = .9 
print("Probability of correct diagnosis (non-sufferer): " + str(correct_diagnosis_non))
misclassify = (.1 * .995) + (.02*.005)
print("Probability of misclassifying someone: " + str(misclassify))

Probability of positive result: 0.1044
Probability of correct diagnosis (sufferer): 0.98
Probability of correct diagnosis (non-sufferer): 0.9
Probability of misclassifying someone: 0.09960000000000001


In [5]:
# Probabilities using Baye's rule
# Probability of having Thripshaws disease given a positive diagnosis
# P(having disease | positive diagnosis) = P(positive diagnosis | having disease) * P(having disease) / P(positive diagnosis)
p1 = .98 * .005 / p_positive
print("Probability of having the disease given a positive diagnosis: %.5f" %p1)

#P(healthy | positive diagnosis) = P(positive diagnosis | healthy) * P(healthy) / P(positive diagnosis)
p2 = .1 * (1-.005) / p_positive
print("Probability of being healthy given a positive diagnosis: %.5f" %p2)

Probability of having the disease given a positive diagnosis: 0.04693
Probability of being healthy given a positive diagnosis: 0.95307


## Results:
     Although the test will correctly identify 98% of the people who actually have Thripshaws, the false positives skew the overal rates and your probability of having Thripshaws given a positive diagnosis is 4.69%. 